contracts/Terms.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity >=0.7.0 <0.9.0; 
+
+contract Terms {
+
+    string taskDescription;
+
+    // Unix timestamp for expiry date
+    uint256 expiry; 
+
+    // TODO: Support > 1 employers pooling their funds together
+    address employerAddr;
+
+    // TODO: Support > 1 contractors receiving payment
+    address contractorAddr; 
+
+    PaymentTerms paymentTerms;
+
+    ConsensusType consensusType;
+
+}
+
+
+/**
+    Payment terms is an interface describing how the employee will be paid.
+ */
+abstract contract PaymentTerms {
+
+    function getPaymentAmount() public virtual view returns(uint);
+
+}
+
+contract FlatFee is PaymentTerms {
+
+    // TODO: How can we generalize this to other tokens and stable coins?
```

Comment:
In Solidity, we can probably only use ETH as a transaction gas. We might be able to write up some contracts that allows us to use values that represent other cryptocurrencies. That's a good question..

Reply:
Got it -- someone else mentioned using `payment streams` and how that could allow you to send and receive erc20 tokens like stable coins. If we have enough time, could be cool to come back and see what's involved.